maven-core/src/main/java/org/apache/maven/internal/impl/PropertiesAsMap.java

```diff
@@ -67,28 +67,30 @@ public boolean hasNext()
                     @Override
                     public Entry<String, String> next()
                     {
-                        if ( next == null )
+                        if ( !hasNext() )
```

Comment:
although functionally correct I still find it weird to change the state in `hasNext()`. Why not put the logic to look for the next entry in `next` and simplify `hasNext()` with a simple check if `next == null` Also I find it very confusing that `next = null` when the end of the iterator has not yet been reached

Reply:
Fixed